Applications/Spire/Include/Spire/Ui/DropDownItem.hpp

```diff
@@ -39,23 +39,23 @@ namespace Spire {
       DropDownItem(QVariant value, QImage icon, QWidget* parent = nullptr);
 
       //! Returns the item's value.
-      const QVariant& get_value() const;
+      virtual const QVariant& get_value() const;
 
       //! Returns true iff the item is highlighted.
-      bool is_highlighted() const;
+      virtual bool is_highlighted() const;
 
       //! Displays the item's highlight style.
-      void set_highlight();
+      virtual void set_highlight();
 
       //! Removes the item's highlight style.
-      void reset_highlight();
+      virtual void reset_highlight();
 
       //! Connects a slot to the highlighted signal.
-      boost::signals2::connection connect_highlighted_signal(
+      virtual boost::signals2::connection connect_highlighted_signal(
         const HighlightedSignal::slot_type& slot) const;
       
       //! Connects a slot to the selected signal.
-      boost::signals2::connection connect_selected_signal(
+      virtual boost::signals2::connection connect_selected_signal(
```

Comment:
Why are these now virtual? In general I do not like to see changes to an API that have not been discussed/approved.

Reply:
I subclass the DropDownItem for items that require custom widgets or painting. Doesn't that require the base class to have virtual methods when used polymorphically?

Reply:
I would argue they should have been virtual to begin with, but the other widget that subclasses the DropDownItem, SecurityInfoItem, doesn't override any of the base class methods (aside from the protected Qt event functions).

Reply:
Don't need these to be ```virtual```.

Reply:
Why?

Reply:
The default is non-```virtual```, if method is declared to be ```virtual``` there needs to be a justification but I don't see one.

Reply:
So is the standard we declare functions as virtual on an as-needed basis?

Reply:
Yep! That's exactly it.